### PR TITLE
FIx issues

### LINKED ISSUE #94: Entity vehicle variable.
The current name for what a user is currently riding in the Entity class is vehicle, I don't think this really makes much sense, I've never really heard anyone refer to things like horses or pigs as vehicles. I can make a PR for ridingEntity, mountedEntity, or something along those lines if anyone agrees.

### LINKED ISSUE #121: "tick" or "update" and when to use them?

### LINKED ISSUE #170: Issues adding argument mappings to Block.createCuboid

### LINKED ISSUE #171: Make sense of .util.shape.*
Some of the classes are in, but I'm not sure what the DoubleLists are for. Can someone with a better working knowledge of the area chime in?

### LINKED ISSUE #297: Colorful Mapping Proposals
- [ ] `ItemColorMap` -> `ItemColors`
- [ ] `ItemColorMapper` -> `ItemColorProvider`
- [ ] `BlockColorMap` -> `BlockColors`
- [ ] `BlockColorMapper` -> `BlockColorProvider`
- [ ] `method_1704` -> `getColor`
- [ ] `method_1708` -> `register`

### LINKED ISSUE #384: "darkness" vs "subtracted light"
>Iirc darkness is used in other places for similar function (specifically World operations). Should this be made consistent?

https://github.com/FabricMC/yarn/pull/380

### LINKED ISSUE #476: net.minecraft.container.AnvilContainer.pos wrong name
`net.minecraft.container.AnvilContainer.pos` (`field_7770`) does not denote a position.
It is a Property type, and it is used to store the level cost for combining 2 items in the GUI.

### LINKED ISSUE #499: Block.friction and related methods/fields are actually the opposite
The `Block.friction` field and all its related fields/methods (`Block.Settings.friction`, etc) actually work such that a higher value means *less friction* and a lower value means *more friction*. For example, ice sets this field to 0.98 while the default is set to 0.6.

Can't really come up with too many good names. Maybe `frictionlessness`? Any one have better ideas?

### LINKED ISSUE #552: Issue with MemoryModuleType.GOLEM_SPAWN_CONDITIONS mapping

### LINKED ISSUE #571: net.minecraft.world.chunk.ChunkPos -> net.minecraft.util.math.ChunkPos
To group it with `BlockPos`, `ChunkSectionPos`, and other position-related classes.

### PR DESCRIPTION
**Pull request by Runemoro**
I went through the yarn issues and fixed most bad names, but avoided the more controversial changes. This PR fixes:

 - Closes https://github.com/FabricMC/yarn/issues/678
 - Closes https://github.com/FabricMC/yarn/issues/676
 - Closes https://github.com/FabricMC/yarn/issues/668
 - Closes https://github.com/FabricMC/yarn/issues/666
 - Closes https://github.com/FabricMC/yarn/issues/665
 - Closes https://github.com/FabricMC/yarn/issues/656
 - Closes https://github.com/FabricMC/yarn/issues/654
 - Closes https://github.com/FabricMC/yarn/issues/637
 - Closes https://github.com/FabricMC/yarn/issues/634
 - Closes https://github.com/FabricMC/yarn/issues/619
 - Closes https://github.com/FabricMC/yarn/issues/594
 - Closes https://github.com/FabricMC/yarn/issues/571
 - Closes https://github.com/FabricMC/yarn/issues/499
 - Closes https://github.com/FabricMC/yarn/issues/121
 - Closes https://github.com/FabricMC/yarn/issues/121

Additionally, these issues seem to have been fixed in previous PRs, or don't seem to be an issue anymore:

 - Closes https://github.com/FabricMC/yarn/issues/552
 - Closes https://github.com/FabricMC/yarn/issues/476
 - Closes https://github.com/FabricMC/yarn/issues/384
 - Closes https://github.com/FabricMC/yarn/issues/297
 - Closes https://github.com/FabricMC/yarn/issues/171
 - Closes https://github.com/FabricMC/yarn/issues/170
 - Closes https://github.com/FabricMC/yarn/issues/94

### CODE REVIEW

**Review by modmuss50 — APPROVED** [review]
Looks good to me, nice load of issues closed with this one.

**Comment by modmuss50** [comment]
Was just about to point out that duplicate class ^ ;)

File: mappings/net/minecraft/entity/LivingEntity.mapping [hunk 1]
```diff
@@ -187,7 +187,7 @@ CLASS aio net/minecraft/entity/LivingEntity
 	METHOD d setSleepingPosition (Lev;)V
 	METHOD da getMovementSpeed ()F
 	METHOD db tickNewAi ()V
-	METHOD dc doPushLogic ()V
+	METHOD dc tickEntityPushing ()V
```
Comment (NeunEinser) [inline mappings/net/minecraft/entity/LivingEntity.mapping]: `tickPushing`?

Comment (liach) [reply mappings/net/minecraft/entity/LivingEntity.mapping]: `tickPushLogic`

**Review by liach — APPROVED** [review]

**Review by NeunEinser — APPROVED** [review]

**Comment by NeunEinser** [comment]
Looks really good, nice work!

**Comment by liach** [comment]
@modmuss50 consider merging? This `Unit` rename helps me so much.

**Pull request merged**